pkg/controller/directimagemigration/validation.go

```diff
@@ -0,0 +1,265 @@
+	migref "github.com/konveyor/mig-controller/pkg/reference"
+	kapi "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Types
+const (
+	InvalidSourceClusterRef               = "InvalidSourceClusterRef"
+	InvalidDestinationClusterRef          = "InvalidDestinationClusterRef"
+	InvalidSourceCluster                  = "InvalidSourceCluster"
+	InvalidDestinationCluster             = "InvalidDestinationCluster"
+	SourceClusterNotReady                 = "SourceClusterNotReady"
+	DestinationClusterNotReady            = "DestinationClusterNotReady"
+	MissingSourceClusterRegistryPath      = "MissingSourceClusterRegistryPath"
+	MissingDestinationClusterRegistryPath = "MissingDestinationClusterRegistryPath"
+	NsListEmpty                           = "NamespaceListEmpty"
+	NsNotFoundOnSourceCluster             = "NamespaceNotFoundOnSourceCluster"
+
+	Running   = "Running"
+	Failed    = "Failed"
+	Succeeded = "Succeeded"
+)
+
+// Reasons
+const (
+	NotFound    = "NotFound"
+	NotSet      = "NotSet"
+	NotDistinct = "NotDistinct"
+	NotReady    = "NotReady"
+)
+
+// Categories
+const (
+	Critical = migapi.Critical
+	Advisory = migapi.Advisory
+)
+
+// Statuses
+const (
+	True  = migapi.True
+	False = migapi.False
+)
+
+// Validate the image migration resource
+func (r ReconcileDirectImageMigration) validate(imageMigration *migapi.DirectImageMigration) error {
+	err := r.validateSrcCluster(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	err = r.validateDestCluster(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	// Migrated namespaces.
+	err = r.validateNamespaces(imageMigration)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	return nil
+}
+
+func (r ReconcileDirectImageMigration) validateSrcCluster(imageMigration *migapi.DirectImageMigration) error {
+	ref := imageMigration.Spec.SrcMigClusterRef
+
+	// Not Set
+	if !migref.RefSet(ref) {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidSourceClusterRef,
+			Status:   True,
+			Reason:   NotSet,
+			Category: Critical,
+			Message:  "The directImageMigration.srcMigClusterRef must reference name and namespace for a valid `MigCluster`",
+		})
+		return nil
+	}
+
+	cluster, err := migapi.GetCluster(r, ref)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	// Not found
+	if cluster == nil {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     InvalidSourceClusterRef,
+			Status:   True,
+			Reason:   NotFound,
+			Category: Critical,
+			Message: fmt.Sprintf("The directImageMigration.srcMigClusterRef %s must reference a valid `MigCluster`",
+				path.Join(imageMigration.Spec.SrcMigClusterRef.Namespace, imageMigration.Spec.SrcMigClusterRef.Name)),
+		})
+		return nil
+	}
+
+	// Not ready
+	if !cluster.Status.IsReady() {
+		imageMigration.Status.SetCondition(migapi.Condition{
+			Type:     SourceClusterNotReady,
+			Status:   True,
+			Category: Critical,
```

Comment:
NIT: the reason field is missing in the condition? I'm not sure if this is intentional...

Reply:
Not intentional. Apparently the migplan validation code I pulled this from was missing it. I'll add it here.